app/javascript/packages/verify-flow/steps/password-confirm/password-reset-button.tsx

```diff
@@ -0,0 +1,39 @@
+import { SpinnerButton } from '@18f/identity-spinner-button';
+import { t } from '@18f/identity-i18n';
+import { isErrorResponse, post } from '../../services/api';
+import type { ErrorResponse } from '../../services/api';
+
+/**
+ * API endpoint for password reset.
+ */
+export const API_ENDPOINT = '/api/verify/v2/password_reset';
+
+/**
+ * API response shape.
+ */
+interface PasswordResetSuccessResponse {
+  redirect_url: string;
+}
+
+/**
+ * API response shape.
+ */
+type PasswordResetResponse = PasswordResetSuccessResponse | ErrorResponse;
+
+function PasswordResetButton() {
+  async function requestReset() {
+    const json = await post<PasswordResetResponse>(API_ENDPOINT, {}, { csrf: true, json: true });
+    if (!isErrorResponse(json)) {
+      const { redirect_url: redirectURL } = json;
+      window.location.href = redirectURL;
```

Comment:
I feel like I'm gonna forget and ask this every time, but should we inject some sort of callback so we can avoid stubbing `window.location` in tests like we did for `onComplete` inside the `VerifyFlow`?

Reply:
> I feel like I'm gonna forget and ask this every time, but should we inject some sort of callback so we can avoid stubbing `window.location` in tests like we did for `onComplete` inside the `VerifyFlow`?

Hm, yeah, I like the idea of this sort of dependency injection. Would you imagine it as some context callback that takes any ol' URL to redirect to?

Another option could be to specify it as a prop with a default value used in the browser, and assigned a simple stub in specs ([we do similar in DocAuth `UploadContextProvider`](https://github.com/18F/identity-idp/blob/35a15afed0c9ed9604e3a66e9865b29e35c09388/app/javascript/packages/document-capture/context/upload.jsx#L85)).

```tsx
const navigate = (url) => window.location.href = url;

function PasswordResetButton({ onNavigate = navigate }) {
  // ...
}
```

Reply:
Went with prop idea in c458ec68d.

Reply:
> Considering that we do essentially the same thing in a bunch of files, might be worth putting in its own constant (so we could do `onNavigate = DEFAULT_NAVIGATE` or something? But that also might not be worth the indirection for such simple functionality

Yeah I think my key concern there would just not having a great and obvious place to put it (might also contribute to the indirection problem). Let's revisit if it starts becoming a pain to repeat.